### PR TITLE
chore(remap): Remove default args from functions

### LINKED ISSUE #5745: Remove default args from VRL functions
At the moment, there are a few VRL functions that enable you to supply default values as fallbacks, such as several of the `to_*` functions. These were added pretty early in the life of the language. Since then, however, it's been decided that supplying defaults this way is both unnecessary, as there are other language constructs like `||` for defaulting to a value, and undesirable, as it requires explicitly building this into every function (and testing, etc.). These fallback-style default args should be removed from the current slate of functions.

### PR DESCRIPTION
**Pull request by lucperkins**
Closes #5745. This PR removes `default` args from all VRL functions that have them. The reasoning is that there will be constructs in the language for assigning default values, meaning that baking a defaulting mechanism into each function is sub-optimal both in terms of UX and development/maintenance workload.

### CODE REVIEW

File: lib/remap-functions/src/parse_json.rs [hunk 1]
```diff
@@ -9,63 +9,41 @@ impl Function for ParseJson {
...
-            Ok(value)
-        };
-
-        crate::util::convert_value_or_default(
-            self.value.execute(state, object),
-            self.default.as_ref().map(|v| v.execute(state, object)),
-            to_json,
-        )
+        let value = self.value.execute(state, object)?;
+        let bytes = value.unwrap_bytes();
```
Comment (JeanMertz) [inline lib/remap-functions/src/parse_json.rs:34]:
```suggestion
        let bytes = value.try_bytes()?;
```

I know you didn't change this in this PR, but we should avoid unwrapping if we aren't 100% sure this will always be bytes.

File: lib/remap-functions/src/parse_timestamp.rs [hunk 2]
```diff
@@ -21,106 +21,56 @@ impl Function for ParseTimestamp {
...
-        };
+        use value::Kind;
 
-        value_def
-            .merge_with_default_optional(default_def)
-            .merge_optional(format_def)
-            .with_constraint(value::Kind::Timestamp)
+        self.value
+            .type_def(state)
+            .fallible_unless(Kind::Bytes | Kind::Timestamp)
```
Comment (JeanMertz) [inline lib/remap-functions/src/parse_timestamp.rs:72]: I believe this function is always fallible, since we have to parse the `format` string at runtime.

File: lib/remap-functions/src/to_int.rs [hunk 3]
```diff
@@ -12,7 +12,7 @@ impl Function for ToInt {
     fn parameters(&self) -> &'static [Parameter] {
         &[Parameter {
             keyword: "value",
-            accepts: |v| matches!(v, Value::Integer(_) | Value::Float(_) | Value::Bytes(_) | Value::Boolean(_) | Value::Timestamp(_) | Value::Null),
+            accepts: crate::util::is_scalar_value,
```
Comment (JeanMertz) [inline lib/remap-functions/src/to_int.rs:15]: `is_scalar_value` does not return true for `Value::Timestamp(_)`, so I think we'll have to revert this change? (that function is poorly named, it started out correctly, but has since diverged from its name a bit)

File: lib/remap-functions/src/to_string.rs [hunk 4]
```diff
@@ -58,21 +50,12 @@ impl Expression for ToStringFn {
...
+        }
     }
 
     fn type_def(&self, state: &state::Compiler) -> TypeDef {
         self.value
             .type_def(state)
-            .merge_with_default_optional(
-                self.default.as_ref().map(|default| default.type_def(state)),
-            )
             .with_constraint(value::Kind::Bytes)
```
Comment (JeanMertz) [inline lib/remap-functions/src/to_string.rs:59]: This has to have an `fallible_unless` line, since maps and arrays always make this function fallible.

**Review by JeanMertz — CHANGES_REQUESTED** [review]
A couple of issues left to fix, but this is looking good 👌

**Review by FungusHumungus — APPROVED** [review]
